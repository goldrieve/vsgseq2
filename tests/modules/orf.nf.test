nextflow_process {

    name "Test Process ORF"
    script "modules/orf.nf"
    process "ORF"

    test("Should identif ORFs") {

        when {
            params {
                cdslength = 300
                partial = false
            }
            process {
                """
                input[0] = "$projectDir/tests/data/latesingle_trinity_cdhit.fasta"
                input[1] = params.cdslength
                input[2] = params.partial
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
