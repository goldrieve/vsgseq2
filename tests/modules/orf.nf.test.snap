{
    "Should identif ORFs": {
        "content": [
            {
                "0": [
                    "latesingle_ORF.fasta:md5,4088d10ae89a693a2f5d05b51fb642f5"
                ],
                "fasta": [
                    "latesingle_ORF.fasta:md5,4088d10ae89a693a2f5d05b51fb642f5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-21T15:19:42.02271792"
    }
}